### PR TITLE
[receiver/splunkhec] Metrics dropped when logs used

### LINKED ISSUE #27473: [receiver/splunkhec] if a receiver is used for metrics and logs, metrics are lost
### Component(s)

receiver/splunkhec

## Description
The splunkhecreceiver loses metrics sent to it when used for both metrics and logs.

## Steps to Reproduce
Set up the same receiver for metrics and logs (used in a pipeline in both metrics and logs pipeline)
Send metric data to the receiver.

## Expected Result
Metrics should make past the receiver.

## Actual Result
No metrics are consumed.

### Collector version

0.86.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Fix metrics dropped when logs and metrics used at the same time

**Link to tracking Issue:**
Fixes #27473 

**Testing:**
Unit tests.

### CODE REVIEW

File: receiver/splunkhecreceiver/receiver.go [hunk 1]
```diff
@@ -58,11 +58,11 @@ var (
 	errInvalidMethod          = errors.New("invalid http method")
 	errInvalidEncoding        = errors.New("invalid encoding")
 
-	okRespBody                = initJSONResponse(responseOK)
```
Comment (fatsheep9146) [inline receiver/splunkhecreceiver/receiver.go:61]: Why not use `initJSONResponse` any more?

File: receiver/splunkhecreceiver/receiver_test.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ import (
 	"time"
 
 	jsoniter "github.com/json-iterator/go"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/sharedcomponent"
```
Comment (fatsheep9146) [inline receiver/splunkhecreceiver/receiver_test.go:21]: I think this import should move down, next to `"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/testutil"`

File: receiver/splunkhecreceiver/receiver.go (hunk 1 above)
Comment (atoulme) [reply receiver/splunkhecreceiver/receiver.go:61]: It doubles encodes what is now a valid JSON object into a JSON string.

File: receiver/splunkhecreceiver/receiver_test.go (hunk 2 above)
Comment (atoulme) [reply receiver/splunkhecreceiver/receiver_test.go]: thanks, fixing

File: receiver/splunkhecreceiver/receiver.go (hunk 1 above)
Comment (atoulme) [reply receiver/splunkhecreceiver/receiver.go:61]: Sigh. I should decouple this into another PR.

File: receiver/splunkhecreceiver/receiver.go [hunk 3]
```diff
@@ -336,85 +337,77 @@ func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request)
...
-				r.failRequest(ctx, resp, http.StatusBadRequest, []byte(fmt.Sprintf(responseErrHandlingIndexedFields, len(events))), len(events), nil)
+				r.failRequest(ctx, resp, http.StatusBadRequest, []byte(fmt.Sprintf(responseErrHandlingIndexedFields, len(events)+len(metricEvents))), len(events)+len(metricEvents), nil)
 				return
 			}
 		}
 		if msg.IsMetric() {
 			if r.metricsConsumer == nil {
-				r.failRequest(ctx, resp, http.StatusBadRequest, errUnsupportedMetricEvent, len(events), err)
+				r.failRequest(ctx, resp, http.StatusBadRequest, errUnsupportedMetricEvent, len(metricEvents), err)
 				return
```
Comment (fatsheep9146) [inline receiver/splunkhecreceiver/receiver.go:369]: I am a little confused about this error reason `errUnsupportedMetricEvent`, when I first look at this, I thought it means that the metric event I sent is not legal(for example the format is not legal), WDYT? Should we change another expression?

Comment (atoulme) [reply receiver/splunkhecreceiver/receiver.go:369]: it's sent if a metric is sent when the receiver is not on a metric pipeline.

File: receiver/splunkhecreceiver/receiver.go (hunk 1 above)
Comment (atoulme) [reply receiver/splunkhecreceiver/receiver.go]: done.

**Review by fatsheep9146 — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**